src/library/Uno.Material/MaterialColors.cs

```diff
@@ -28,18 +28,35 @@ public string OverrideSource
 		private static void OnColorPaletteOverrideSourceChanged(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs args)
 		{
 			ColorPaletteOverrideSource = args.NewValue as string;
+
+			ApplyOverride();
 		}
 
-		public MaterialColors()
+		internal static void ApplyOverride(ResourceDictionary target = null)
 		{
-			MergedDictionaries.Add(new ResourceDictionary { Source = new Uri("ms-appx:///Uno.Material/Styles/Application/ColorPalette.xaml") });
 			if (!string.IsNullOrWhiteSpace(ColorPaletteOverrideSource))
 			{
-				MergedDictionaries.Add(new ResourceDictionary { Source = new Uri(ColorPaletteOverrideSource) });
-			}
+				var overrideDictionary = new ResourceDictionary { Source = new Uri(ColorPaletteOverrideSource) };
+
+				target = target ?? Application.Current?.Resources;
+
+				if (target != null)
+				{
+					// The Unsource part is needed to be able to overwrite values in UWP. This target does not
+					// allow for new values to be provided when a ResourceDictionary is initialized using the
+					// `Source` property. The Unsource method finds the first reference to the specific ResourceDictionary
+					// clones it, then replaces the original one so the `OverwriteValues` method change take the values
+					// provided in the ColorPaletteOverrideSource dictionary.
 
-			InitializeComponent();
+					Uno.Themes.ResourceDictionaryHelper.Unsource(target, new Uri("ms-appx:///Uno.Material/Styles/Application/ColorPalette.xaml"));
```

Comment:
if it's only necessary on UWP, would it improve perf on other platforms to make it conditional?

Reply:
Indeed.

Reply:
This is now removed, as unused.